pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -57,17 +57,19 @@ func ValidateCertificateSpec(crt *cmapi.CertificateSpec, fldPath *field.Path) fi
 		el = append(el, validateEmailAddresses(crt, fldPath)...)
 	}
 
-	switch crt.KeyAlgorithm {
-	case "", cmapi.RSAKeyAlgorithm:
-		if crt.KeySize > 0 && (crt.KeySize < 2048 || crt.KeySize > 8192) {
-			el = append(el, field.Invalid(fldPath.Child("keySize"), crt.KeySize, "must be between 2048 & 8192 for rsa keyAlgorithm"))
-		}
-	case cmapi.ECDSAKeyAlgorithm:
-		if crt.KeySize > 0 && crt.KeySize != 256 && crt.KeySize != 384 && crt.KeySize != 521 {
-			el = append(el, field.NotSupported(fldPath.Child("keySize"), crt.KeySize, []string{"256", "384", "521"}))
+	if crt.PrivateKey != nil {
+		switch crt.PrivateKey.Algorithm {
+		case "", cmapi.RSAKeyAlgorithm:
```

Comment:
Doesn't the defaulting and validation ensure that this field always has a non-empty value?

Reply:
Nope, this field is `optional` meaning a user is allowed to submit an empty value to the API (i.e. it will pass validation), but we don't perform defaulting at the moment because of https://github.com/jetstack/cert-manager/pull/3038#discussion_r448964935